Remove native_functions.yaml dependency from some reduction operators

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64499
* __->__ #64173
* #64172
* #64171
* #64170

This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

Differential Revision: [D30728581](https://our.internmc.facebook.com/intern/diff/D30728581)

peterbell10 committed:
Remove native_functions.yaml dependency from some reduction operators

This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30728581).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

Differential Revision: [D30728581](https://our.internmc.facebook.com/intern/diff/D30728581)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

Differential Revision: [D30728581](https://our.internmc.facebook.com/intern/diff/D30728581)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

Differential Revision: [D30728581](https://our.internmc.facebook.com/intern/diff/D30728581)

[ghstack-poisoned]

ezyang commented on aten/src/ATen/core/Tensor.cpp:
```diff
@@ -15,6 +15,18 @@ TensorBase TensorBase::__dispatch_contiguous(c10::MemoryFormat memory_format) co
+  Tensor self(*this);
+  at::_ops::zero_::call(self);
+  return *this;
+}
```
Special dispensation, I see...

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

Differential Revision: [D30728581](https://our.internmc.facebook.com/intern/diff/D30728581)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

Differential Revision: [D30728581](https://our.internmc.facebook.com/intern/diff/D30728581)

[ghstack-poisoned]

ezyang committed:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

Differential Revision: [D30728581](https://our.internmc.facebook.com/intern/diff/D30728581)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30728581).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

Differential Revision: [D30728581](https://our.internmc.facebook.com/intern/diff/D30728581)

[ghstack-poisoned]

ezyang committed:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

Differential Revision: [D30728581](https://our.internmc.facebook.com/intern/diff/D30728581)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30728581).

Commit:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

Differential Revision: [D30728581](https://our.internmc.facebook.com/intern/diff/D30728581)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

Differential Revision: [D30728581](https://our.internmc.facebook.com/intern/diff/D30728581)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

Differential Revision: [D30728581](https://our.internmc.facebook.com/intern/diff/D30728581)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

Differential Revision: [D30728581](https://our.internmc.facebook.com/intern/diff/D30728581)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

Differential Revision: [D30728581](https://our.internmc.facebook.com/intern/diff/D30728581)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30728581).

Commit:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

Differential Revision: [D30728581](https://our.internmc.facebook.com/intern/diff/D30728581)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

Differential Revision: [D30728581](https://our.internmc.facebook.com/intern/diff/D30728581)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from some reduction operators"


This one also required restructuring the code a bit to move the kernel
code into seperate files. So, I've mainly focused on CUDA which is
where the real build-time issues are.

Differential Revision: [D30728581](https://our.internmc.facebook.com/intern/diff/D30728581)

[ghstack-poisoned]
